src/main/resources/db/migration/V1_68__create_utilisation_report.sql

```diff
@@ -0,0 +1,313 @@
+	on pool_members.juror_number = appearance.juror_number and pool_members.report_date = appearance.report_date
+	group by pool_members.juror_number,
+	pool_members.report_date,
+	pool_members.service_from,
+ 	pool_members.service_to,
+ 	panel_members.trial,
+ 	panel_members.jury_from,
+	panel_members.jury_to,
+ 	pool_members.report_date,
+	pool_members.within_service,
+	panel_members.active_trial,
+	appearance.attendance,
+	pool_members.holiday
+	order by pool_members.juror_number, pool_members.report_date;
+
+END;
+$function$
+;
+
+
+-- function to return jurors and service within the report window (also indicating public holidays)
+CREATE OR REPLACE FUNCTION juror_mod.util_report_juror_service_list(p_loc_code text, p_start_date date, p_end_date date)
+ RETURNS TABLE(
+ 	juror_number character varying,
+	report_from date,
+	report_to date,
+	service_from date,
+	service_to date,
+	report_date date,
+	within_service boolean,
+	holiday boolean)
+ LANGUAGE plpgsql
+AS $function$
+begin
+
+return query select
+	pool_members.juror_number,
+	p_start_date as report_from,
+	p_end_date as report_to,
+ 	pool_members.service_start_date as service_from,
+ 	pool_members.service_end_date as service_to,
+ 	report_days.report_date,
+	case when report_days.report_date between pool_members.service_start_date and pool_members.service_end_date then true else false end as within_service,
+	report_days.holiday
+	from (select * from juror_mod.util_report_report_days_list(p_loc_code, p_start_date, p_end_date)) as report_days
+	left join (select * from juror_mod.util_report_pool_members_list(p_loc_code, p_start_date, p_end_date)) as pool_members
+	on report_days.report_date between pool_members.service_start_date and pool_members.service_end_date
+	group by  pool_members.juror_number, report_days.report_date, report_days.holiday, pool_members.service_start_date, pool_members.service_end_date
+	order by pool_members.juror_number, report_date;
+END;
+$function$
+;
+
+
+-- function to get the list of jurors in a court within a given date range (used in function above)
+CREATE OR REPLACE FUNCTION juror_mod.util_report_pool_members_list(p_loc_code text, p_start_date date, p_end_date date)
+ RETURNS TABLE(juror_number character varying, attendance_date date, return_date date, service_start_date date, service_end_date date, min_status integer)
+ LANGUAGE plpgsql
+AS $function$
+begin
+
+return query select
+	jp.juror_number,
+    min(a.attendance_date) attendance_date,
+    min(case when jp.is_active = true then p.return_date else to_date('01/12/2099', 'dd/mm/yyyy') end ) return_date,
+    greatest(least(coalesce(min(a.attendance_date), min(case when jp.is_active = true then p.return_date else to_date('01/12/2099', 'dd/mm/yyyy') end )),
+    		min(case when jp.is_active = true then p.return_date else to_date('01/12/2099','dd/mm/yyyy') end )), p_start_date) service_start,
```

Comment:
gonna need a comment here to explain this 😅 I managed to work through it and understand what it's doing, and it looks right 👌 but I know if I come back to it I'll have to work through it again 😛 

we're using the future default date because migrated appearance records may not have a pool_number to lookup the pool return date - good catch! 👍

Reply:
good point, have added a comment now